dds/idl/typeobject_generator.cpp

```diff
@@ -0,0 +1,427 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "typeobject_generator.h"
+#include "be_extern.h"
+
+#include "utl_identifier.h"
+
+#include "topic_keys.h"
+
+using std::string;
+using namespace AstTypeClassification;
+
+namespace {
+
+string tag_type(UTL_ScopedName* name)
+{
+  return dds_generator::scoped_helper(name, "_") + "_xtag";
+}
+
+void
+call_get_minimal_type_identifier(AST_Type* type)
```

Comment:
This is based on Minimal Type Identifiers and Objects.  Do we need Completes in this iteration?